src/main/Numbers.java

```diff
@@ -0,0 +1,40 @@
+package main;
+
+import java.util.Collections;
+import java.util.List;
+
+public class Numbers<T extends Number> {
+    private List<T> numbersList;
+
+    public Numbers(List<T> numbersList) {
+        this.numbersList = numbersList;
+    }
+
+    public int countEvenNumbers() {
+        int counter = 0;
+        for (T number : numbersList) {
+            if (Math.abs(number.intValue()) % 2 == 0) {
```

Comment:
Do you think `4.3` is an even number?
What is the result of `new Numbers(Arrays.asList(4.1, 4.2, 4.3)).countEvenNumbers()`?